### PR TITLE
Adding executable code for `tf.sparse.softmax`

### PR DESCRIPTION
**Pull request by sushreebarsa**
Changed the documentation of tf.sparse.softmax with executable example code.
The values passed to tf.sparse.SparseTensor must be a rank-1 tensor instead of a rank-3 tensor.

Fixes #[55035](https://github.com/tensorflow/tensorflow/issues/55035)

### CODE REVIEW

**Comment by mihaimaruseac** [comment]
Please use a proper title, not "Update <file>"

https://cbea.ms/git-commit/

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Can you make these testable docstrings?

https://www.tensorflow.org/community/contribute/docs_ref#testable_docstrings

File: tensorflow/python/ops/sparse_ops.py [hunk 1]
```diff
@@ -2675,13 +2675,21 @@ def sparse_softmax(sp_input, name=None):
...
 
   ```python
-  import tensorflow as tf
-  shape = [2, 2, 2]  # 3-D SparseTensor
-  values = np.asarray([[[0., np.e], [1., 0.]], [[np.e, 0.], [np.e, np.e]]])
-  indices = np.vstack(np.where(values)).astype(np.int64).T
-  values = values[np.where(values)] # Flatten values
-  result = tf.sparse.softmax(tf.sparse.SparseTensor(indices, values, shape)) 
-  print(tf.sparse.to_dense(result))
+  >>> import tensorflow as tf
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/sparse_ops.py:2678]: You don't need this line.

Also, you don't need the triple-back-quote lines above and at the end

**Comment by gbaned** [comment]
Hi @sushreebarsa  Can you please check @mihaimaruseac's comments and keep us posted ? Thank you!

**Comment by sushreebarsa** [comment]
@gbaned Requested changes have been updated. Could you please review and let me know for any further modifications.
Thank you!

File: tensorflow/python/ops/sparse_ops.py [hunk 2]
```diff
@@ -2675,13 +2675,10 @@ def sparse_softmax(sp_input, name=None):
   Example:
 
   ```python
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/sparse_ops.py:2677]: You don't need this either

File: tensorflow/python/ops/sparse_ops.py [hunk 3]
```diff
@@ -2674,24 +2674,15 @@ def sparse_softmax(sp_input, name=None):
...
-  ```
-
+  >>> shape = [2, 2, 2]  # 3-D SparseTensor
+  >>> values = np.asarray([[[0., np.e], [1., 0.]], [[np.e, 0.], [np.e, np.e]]])
+  >>> indices = np.vstack(np.where(values)).astype(np.int64).T
+  <tf.Tensor(
+   [[[0.  1. ]
+   [1.  0. ]]
+   [[1.  0. ]
+   [0.5 0.5]]], shape=(2, 2, 2), dtype=float64)>
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/sparse_ops.py:2684]: I don't think this is the right output. Can you make sure you run the doctest?

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/sparse_ops.py (hunk 3 above)
Comment (sushreebarsa) [reply tensorflow/python/ops/sparse_ops.py:2684]: Hi @mihaimaruseac! 
I have changed the code from first cell to second cell as shown in this [gist](https://colab.research.google.com/gist/sushreebarsa/792f4b3f9df002ee48f2e7d98d7a6e23/untitled60.ipynb). Please confirm and let me know if further modifications are required. Thank you!

Comment (mihaimaruseac) [reply tensorflow/python/ops/sparse_ops.py:2684]: Please run the doctest. You will see it failing, because `x = expr` does not output.

Also, this doctest does not seem to use the TF API at all. Please update.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
Hi @sushreebarsa Can you please check @mihaimaruseac's comments. Thank you!

**Comment by gbaned** [comment]
Hi @sushreebarsa Can you please check @mihaimaruseac's comments. Thank you!

**Comment by gbaned** [comment]
Hi @sushreebarsa Can you please check @mihaimaruseac's comments. Thank you!

**Comment by mihaimaruseac** [comment]
Closing since it stalled
